### DISCUSSION TITLE
Shannon Entropy

### DISCUSSION THREAD

**Discussion by sas-b9** [IBAHCM/RPiR#557] [Q&A for project] [answered]
Hello Richard @richardreeve 

you mentioned for shannon entropy to use..

if (abs(q-1) < 0.000001) instead of q=1 which I did within the if statement you said to do in the general diversity function of project 3

however then for my shannon entropy function, nothing works unless I put "q = 0.000001", I tried q=(abs(q-1) < 0.000001) and q=1, however nothing worked?

Lastly you mentioned about making all these different functions, however in the demo you just ask for the general diversity function to be called, sorry I am confused, as I just made all 4 functions and called them individually in the demo...

"The demo should then call the general function directly (not the 4 specific functions) and just calculate the general diversity measures for the datasets provided with q=0, 1, 2, and ∞ "

Any assistance would be appreciated

**Comment by richardreeve** [comment]
On Shannon Entropy - I think you're misunderstanding where the `if (abs(q-1) < 0.000001)` statement goes. It's used to test whether `q` really is equal to 1 in the function instead of the statement `if (q == 1)` in the general function, because `q` may not be exactly 1. You still call the general function with `q=1` in your entropy function.

In the demo, I think what I ask you to do is very clear and you seem to understand it - call the general function in the demo and not the individual diversity functions - but you chose not to do it. You will be marked down if you insist on not doing what I asked you to do, so I recommend following the instructions.

**Reply by sas-b9** [reply]
Thankyou for answering, have you got any reason when I plug in if (abs(q-1) < 0.000001) it does produce a result but just 'TRUE' when I call it with q=1 in my demo of my general diversity function, I understanding it is testing whether it is equal to 1 but I thought this would produce an answer,

sorry for the confusion and many thanks

Many thanks!

**Reply by richardreeve** [reply]
`TRUE` is the result of the calculation `(abs(q-1) < 0.000001)` - that's right. As you say, it's a test to see whether q is 1, and then you still have to write the code to calculate the diversity.

**Comment by mhuzzell** [comment, marked as answer]
It looks to me like the demo for part 3 asks you to call each of the four diversity functions in turn, _and also_ the general diversity function?

> Now write a demo that loads in this new functionality and then calculates the four diversity measures we have discussed for each dataset.
> 
> The demo should _**then**_ call the general function directly (not the 4 specific functions) and just calculate the general diversity measures for the datasets provided with q=0
> , 1, 2, and ∞ (remember, these correspond to the 4 individual diversity measures we have investigated so far). Do you see a progression in the values you observe within each set of population counts as q increases? Comment on this in the report.

**Reply by richardreeve** [reply]
Fair point @mhuzzell! I should just leave the text to speak for itself - I didn't read the whole project 3 demo instructions but just the text that was quoted above when I answered. Apologies. You should do both, but the most important bit is still to call the generic function of the same dataset with different values of q and observe what happens.
